src/main/java/core/dao/car/CarDaoJdbc.java

```diff
@@ -0,0 +1,215 @@
+                        + "AND (d.deleted = false OR d.deleted IS NULL)";
+        Car car = null;
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement getByIdStatement = connection.prepareStatement(select)) {
+            getByIdStatement.setLong(1, id);
+            ResultSet resultSet = getByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                car = DaoUtils.parseToCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(String
+                    .format(ErrorMessages.GET.getMessage(),
+                            Car.class.getSimpleName(), id), e);
+        }
+        return Optional.ofNullable(car);
+    }
+    
+    @Override
+    public List<Car> getAll() {
+        List<Car> returnList = new ArrayList<>();
+        String getAll = "SELECT cars.id as car_id, cars.model as car_model,"
+                        + " cars.manufacturer as manufacturer_id,"
+                        + " m.name as manufacturer_name, m.country as manufacturer_country,"
+                        + " d.id as driver_id, d.name as driver_name,"
+                        + " d.licence_number as driver_licence"
+                        + " FROM cars"
+                        + " JOIN manufacturers m ON m.id = cars.manufacturer"
+                        + " JOIN cars_drivers cd ON cd.\"car_Id\" = cars.id"
+                        + " JOIN drivers d on cd.\"driver_Id\" = d.id"
+                        + " WHERE cars.deleted = false AND (d.deleted = false OR d.deleted IS NULL)"
+                        + " ORDER BY car_id";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement getAllStatement = connection.prepareStatement(getAll)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            if (resultSet.next()) {
+                returnList = getListCarParser(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(String
+                    .format(ErrorMessages.GET_ALL.getMessage(),
+                            Car.class.getSimpleName()), e);
+        }
+        return returnList;
+    }
+    
+    @Override
+    public Car update(Car car) {
+        Manufacturer manufacturer = car.getManufacturer();
+        String update = "UPDATE cars SET model = ?, manufacturer = ?"
+                        + " WHERE id = ? AND deleted = false";
+        try (Connection connection = ConnectionUtils.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(update)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+            removeDrivers(car, connection);
+            insertDrivers(car, connection);
```

Comment:
call these methods after the try block and don't pass the connection there

Reply:
I refer to this.